src/util_cs/mod.rs

```diff
@@ -1,3 +1,70 @@
+use crate::LinearCombination;
+use ff::PrimeField;
+
 pub mod bench_cs;
 pub mod metric_cs;
 pub mod test_cs;
+
+pub type Constraint<Scalar> = (
+    LinearCombination<Scalar>,
+    LinearCombination<Scalar>,
+    LinearCombination<Scalar>,
+    String,
+);
+
+pub trait Comparable<Scalar: PrimeField> {
```

Comment:
please add some documentation here how to use and implemen this

Reply:
Done.